src/lib/define-smart-component.js

```diff
@@ -5,31 +5,37 @@ const META = Symbol('META');
 
 /**
  * @typedef {Element} SmartContainer
- * @param {object} context
+ * @property {Object} context
  */
 
 /**
- * @typedef {object|((property: object) => void)} CallbackOrObject
+ * @typedef {Object|((property: Object) => void)} CallbackOrObject
  */
 
 /**
- * @param {object} definition
+ * @typedef {Object} SmartDefinitionParam
+ * @property {TypeHint} type
+ * @property {boolean} [optional=false]
+ * @template TypeHint
+ */
+
+/**
+ * @param {Object} definition
  * @param {string} definition.selector
- * @param {object} definition.params
+ * @param {{[name: string]: SmartDefinitionParam}} definition.params
  * @param {({
  *   container?: SmartContainer,
  *   component?: Element,
- *   context?: object,
- *   onEvent?: (type: string, listener: (detail: object) => void) => void,
+ *   context?: Object,
+ *   onEvent?: (type: string, listener: (detail: Object) => void) => void,
  *   updateComponent?: (
  *     type: string,
  *     callbackOrObject: CallbackOrObject,
  *   ) => void,
  *   signal?: AbortSignal,
  * }) => void} definition.onContextUpdate
- * @param {AbortSignal} [signal]
  */
-export function defineSmartComponent (definition, signal) {
+export function defineSmartComponent (definition) {
```

Comment:
issue: the signal is actually needed

Let me explain, the idea is that you use `defineSmartComponentCore(definition, signal)` to define a smart component and you can pass it a signal as the second param. When this signal is aborted, the smart component definition is removed from the manager.

We never use it ourselves but it's important that such an action can be aborted/cancelled and it's in the smart manager test suite.

Therefore, `defineSmartComponent` should accept a signal and pass it as is to `defineSmartComponentCore`. I'm not sure we need to do this in this PR, as you want.

Also, since we have one signal to abort the smart definition and one for the component disconnection, we may need two names.

Reply:
I noticed that this parameter is never used. That's why I removed it.